src/state/haproxy_route.py

```diff
@@ -238,13 +252,29 @@ def from_provider(
 
                 backends.append(backend)
 
+            tcp_endpoints: list[HAProxyRouteTcpEndpoint] = []
+            tcp_requirers: HaproxyRouteTcpRequirersData = haproxy_route_tcp.get_data(
+                haproxy_route_tcp.relations
+            )
+            relation_ids_with_invalid_data_tcp = tcp_requirers.relation_ids_with_invalid_data
+            for tcp_requirer in tcp_requirers.requirers_data:
+                if haproxy_route.relations and tcp_requirer.application_data.port in [80, 443]:
```

Comment:
What if two requires request the same port? Don't we need to mark one as invalid?